### PR TITLE
Add hup getter

### PR DESCRIPTION
**Pull request by MikeHathaway**

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
LGTM

**Review by EdNoepel — CHANGES_REQUESTED** [review]
Comments inline.

**Comment by MikeHathaway** [comment]
> We should be iterating downward, that is, to lower prices. The acronyms make this a little confusing: LUP - lowest utilized price HUP - highest utilizable price LUP is always >= HUP

Thanks for clearing that up. Much better understanding now. Pushed up a change that has `getHup` iterate downwards until it reaches a bucket with available quote tokens.

File: src/ERC20Pool.sol [hunk 1]
```diff
@@ -614,6 +615,27 @@ contract ERC20Pool is IPool, Clone {
...
+    function getHup() public view returns (uint256) {
+        uint256 curPrice = lup;
+        while (true) {
+            (uint256 price,, uint256 down, uint256 amount, uint256 debt,,,) = _buckets.bucketAt(curPrice);
+            if (price == down || amount != 0) {
+                break;
+            }
+
+            // check that there are available quote tokens on deposit in down bucket
+            (,,, uint256 downAmount, uint256 downDebt,,,) = _buckets.bucketAt(down);
```
Comment (EdNoepel) [inline src/ERC20Pool.sol]: `downDebt` isn't used anymore, but doesn't harm anything leaving it there.

**Review by EdNoepel — APPROVED** [review]
Much thanks; LGTM.

File: src/ERC20Pool.sol (hunk 1 above)
Comment (MikeHathaway) [reply src/ERC20Pool.sol]: removed

**Pull request merged**